Rakefile

```diff
@@ -18,6 +19,19 @@ def base_cucumber_path
   "artifacts/#{ENV['BROWSER']}/#{ENV['BROWSERSTACK'] == 'true' ? ENV['BROWSERSTACK_CONFIGURATION_OPTIONS'] : 'other'}"
 end
 
+def compare_yaml_hash(yaml_a, yaml_b, context = [])
+  yaml_a.each do |key, value|
```

Comment:
If i'm understanding here, you're wanting to validate the nested structure of the yaml, but only validate the keys right?

Reply:
Yep, want to validate that all the keys match. But not the values obviously as they are translated. Not looking for a perfect check, only a little safeguard. Simpler ways to do this welcome.

Reply:
Actually it seems quite good. Although I saw it's a SO answer that you've based it off. Only thing I got confused by was the name. It's definitely a **very good** way of using clever recursion. I would have suggested `hashdiff` but that computes value difference too using diff-lcs as an algorithm (Which you obviously won't want), so this is a good solution.

If it was me I'd call it something like `ensure_identical_structure(yml_a, yml_b, context = [])` but that's definitely just a personal opinion. Not something I'm fussed on too much

Reply:
[looks around shiftily] definitely did~n't~ copy paste this guvner.